### PR TITLE
refactor: Using just one Context

### PR DESCRIPTION
**Pull request by bennyn**
This PR fixes the issue we have when logging in / doing a logout / logging back in. Because initially we planned to support multiple accounts (contexts) with one API client but now we are going into the direction of having accounts completly seperate from each other so there is no need for a Context map anymore.

I also included to throw an error if you try to login twice (in a row) without logging out before because this usually indicates that  there is a bug in the software using the API client.

### CODE REVIEW

File: src/ts/Client.ts [hunk 1]
```diff
@@ -10,22 +10,22 @@ import {WebSocketClient} from './tcp';
 const buffer = require('./util/buffer');
 
 class Client extends EventEmitter {
-  public auth: {api: AuthAPI} = {
+  public auth: { api: AuthAPI } = {
```
Comment (gregor) [inline src/ts/Client.ts]: Are we changing our code style?

**Comment by gregor** [comment]
In the long run we will still have the requirement to handle multiple contexts. But I guess we are fine for now.

File: dist/demo.js [hunk 2]
```diff
@@ -1,51 +1,84 @@
...
-      .catch((error) => {
-        return client.login(login);
-      })
-      .then((context) => {
+  function initLogoutButton(client) {
+    LOGOUT_BUTTON.onclick = function(event) {
+      event.preventDefault();
+
+      client.logout().then((context) => {
         console.log('Login successful', context);
```
Comment (Yserz) [inline dist/demo.js]: `console.log('Logout successful', context);`

File: dist/demo.js [hunk 3]
```diff
@@ -1,51 +1,84 @@
...
-        submitButton.className = 'valid';
-        submitButton.firstChild.data = "😊";
+
+        LOGIN_BUTTON.className = '';
+        LOGIN_BUTTON.firstChild.data = 'login';
+
+        LOGOUT_BUTTON.className = '';
+
         return client.connect();
```
Comment (Yserz) [inline dist/demo.js]: This fails when logging in, out and in again

**Pull request merged**